app/models/product.rb

```diff
@@ -15,6 +15,9 @@ class Product < ActiveRecord::Base
   has_many   :price_policies
   has_many   :training_requests, dependent: :destroy
 
+  cattr_accessor(:types) { ConfigurableArray.new([Instrument, Item, Service, Bundle]) }
```

Comment:
Yeah, I didn't replace everywhere this could be used, but there are definitely a few more spots where this could be used. And it's really just a decorator, so it's still using the normal Array class. I contemplated monkey-patching Array, but that felt weird.